src/Tizen.NUI/src/internal/Interop/Interop.Accessibility.cs

```diff
@@ -15,39 +15,48 @@
+            [return: MarshalAs(UnmanagedType.U1)]
+            public static extern bool Say(HandleRef jarg1, string jarg2, bool jarg3, IntPtr jarg4);
 
-            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_pause_resume")]
-            public static extern void PauseResume(global::System.Runtime.InteropServices.HandleRef jarg1, bool jarg2);
+            [DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_pause_resume")]
+            public static extern void PauseResume(HandleRef jarg1, bool jarg2);
 
-            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_stop_reading")]
-            public static extern void StopReading(global::System.Runtime.InteropServices.HandleRef jarg1, bool jarg2);
+            [DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_stop_reading")]
+            public static extern void StopReading(HandleRef jarg1, bool jarg2);
 
-            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_suppress_screen_reader")]
-            [return: global::System.Runtime.InteropServices.MarshalAs(global::System.Runtime.InteropServices.UnmanagedType.U1)]
-            public static extern bool SuppressScreenReader(global::System.Runtime.InteropServices.HandleRef jarg1, bool jarg2);
+            [DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_suppress_screen_reader")]
+            [return: MarshalAs(UnmanagedType.U1)]
+            public static extern bool SuppressScreenReader(HandleRef jarg1, bool jarg2);
 
-            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_BridgeEnableAutoInit")]
+            [DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_BridgeEnableAutoInit")]
             public static extern void BridgeEnableAutoInit();
 
-            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_BridgeDisableAutoInit")]
+            [DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_BridgeDisableAutoInit")]
             public static extern void BridgeDisableAutoInit();
 
-            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_IsEnabled")]
-            [return: global::System.Runtime.InteropServices.MarshalAs(global::System.Runtime.InteropServices.UnmanagedType.U1)]
+            [DllImport(NDalicPINVOKE.Lib, EntryPoint = "csharp_dali_accessibility_IsEnabled")]
+            [return: MarshalAs(UnmanagedType.U1)]
             public static extern bool IsEnabled();
+
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+            public delegate void EnabledDisabledSignalHandler();
```

Comment:
Does EnabledDisabled mean this single handler handles both of Enabled and Disabled event?
How about AvailabilityStateChanged ~~ ?

Reply:
Actually this `delegate` here is just a type declaration for a pointer to function that takes void and returns void (i.e. `void (*)(void)` in C/C++), so, technically, it doesn't really handle anything.

But two such function pointers are parameters to `RegisterEnabledDisabledSignalHandler` (which get connected to `Dali::Accessibility::Bridge::EnabledSignal()` and `::DisabledSignal()`, respectively). Its exact signature in `dali-csharp-binder` is `void RegisterEnabledDisabledSignalHandler(void (*enabled)(void), void (*disabled)(void))`.

I would love to remove this type alias altogether, but, to my knowledge, C# does not allow to directly specify a function pointer as a parameter like it was done in `dali-csharp-binder` (in other words, omit the `delegate` declaration), that is, without resorting to using `IntPtr` which I also briefly considered.

Should I change `EnabledDisabledSignalHandler` to something like `VoidToVoidFunctionPointer` instead? This would probably better reflect what this `delegate` really represents.

Reply:
What I really wanted to know is the reason why the delegate has both of Enable and Disable as a name.
But as you said it is used for both of the signal then it is not too weird.